src/hyperion/experiment_plans/oav_snapshot_plan.py

```diff
@@ -0,0 +1,61 @@
+import dataclasses
+from datetime import datetime
+
+from blueapi.core import BlueskyContext, MsgGenerator
+from bluesky import plan_stubs as bps
+from dodal.devices.oav.oav_detector import OAV
+from dodal.devices.oav.oav_parameters import OAVParameters
+from dodal.devices.smargon import Smargon
+
+from hyperion.device_setup_plans.setup_oav import setup_general_oav_params
+from hyperion.parameters.components import WithSnapshot
+from hyperion.parameters.constants import DocDescriptorNames
+from hyperion.utils.context import device_composite_from_context
+
+OAV_SNAPSHOT_GROUP = "oav_snapshot_group"
+
+
+@dataclasses.dataclass
+class OavSnapshotComposite:
+    smargon: Smargon
+    oav: OAV
+
+
+def create_devices(context: BlueskyContext) -> OavSnapshotComposite:
+    return device_composite_from_context(context, OavSnapshotComposite)  # type: ignore
+
+
+def _setup_oav(
+    composite: OavSnapshotComposite,
+    parameters: WithSnapshot,
+    oav_parameters: OAVParameters,
+):
+    yield from setup_general_oav_params(composite.oav, oav_parameters)
+    yield from bps.abs_set(
+        composite.oav.snapshot.directory, str(parameters.snapshot_directory)
+    )
+
+
+def _take_oav_snapshot(
+    composite: OavSnapshotComposite, parameters: WithSnapshot, omega: float
+):
+    yield from bps.abs_set(composite.smargon.omega, omega, group=OAV_SNAPSHOT_GROUP)
+    time_now = datetime.now()
+    filename = f"{time_now.strftime('%H%M%S')}_oav_snapshot_{omega:.0f}"
+    yield from bps.abs_set(composite.oav.snapshot.filename, filename)
+    yield from bps.trigger(composite.oav.snapshot, group=OAV_SNAPSHOT_GROUP)
+    yield from bps.wait(group=OAV_SNAPSHOT_GROUP)
+    yield from bps.create(DocDescriptorNames.OAV_ROTATION_SNAPSHOT_TRIGGERED)
+    yield from bps.read(composite.oav.snapshot)
+    yield from bps.save()
+
+
+def oav_snapshot_plan(
+    composite: OavSnapshotComposite,
+    parameters: WithSnapshot,
+    oav_parameters: OAVParameters,
+    wait: bool = True,
```

Comment:
This wait parameter is here as requested in original issue, but currently unused since we have to wait for the snapshot trigger completion before firing the event. Am open to suggestions to remove it / alternate structuring to make it useful.

Reply:
As it is we can remove this, I agree. If we need to do something more complex async we'll need to do it at device level anyway.